Source/SpiderEye.Core/Bridge/ClientServicesSupport/BridgeClientService.cs

```diff
@@ -0,0 +1,103 @@
+using System;
+using System.Linq;
+using System.Reflection;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace SpiderEye.Bridge.ClientServicesSupport
+{
+    /// <summary>
+    /// A ClientService represents a a service which is implemented by the client.
+    /// </summary>
+    /// <typeparam name="T">Type of the Interface.</typeparam>
+    public class BridgeClientService<T> : DispatchProxy
+    {
+        private readonly string namePrefix = (typeof(T).GetCustomAttribute<BridgeClientServiceAttribute>()?.Name ?? typeof(T).Name) + ".";
+        private WindowCollection windowCollection;
+
+        /// <summary>
+        /// Creates a proxy instance to the bridged client service.
+        /// </summary>
+        /// <param name="sp">The service provider.</param>
+        /// <returns>The created proxy instance.</returns>
+        public static T Create(IServiceProvider sp)
+        {
+            var proxy = Create<T, BridgeClientService<T>>();
+            var clientService = (BridgeClientService<T>)(object)proxy;
+            clientService!.Init(sp.GetRequiredService<WindowCollection>());
```

Comment:
Wouldn't it make more sense to pass the `WindowCollection` as an argument instead of the whole ServiceProvider?

Reply:
Improved.